packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -154,38 +165,57 @@ class CupertinoAlertDialog extends StatelessWidget {
     }
 
     if (content != null) {
-      children.add(new Flexible(
-        fit: FlexFit.loose,
-        child: new Padding(
-          padding: const EdgeInsets.symmetric(horizontal: 20.0),
+      titleContentGroup.add(
+        new Padding(
+          padding: new EdgeInsets.only(
+            left: edgePadding,
+            right: edgePadding,
+            bottom: edgePadding,
+            top: title == null ? edgePadding : 1.0,
+          ),
           child: new DefaultTextStyle(
             style: _kCupertinoDialogContentStyle,
             textAlign: TextAlign.center,
             child: content,
           ),
         ),
-      ));
+      );
     }
 
-    children.add(const SizedBox(height: 22.0));
+    final List<Widget> children = <Widget>[];
+    if (titleContentGroup.isNotEmpty) {
+      children.add(
+        new Flexible(
+          child: new CupertinoScrollbar(
+            child: new ListView(
+              controller: scrollController,
+              shrinkWrap: true,
+              children: titleContentGroup,
+            ),
+          ),
+        ),
+      );
+    }
 
     if (actions != null) {
       children.add(new _CupertinoButtonBar(
         children: actions,
       ));
     }
 
-    return new CupertinoDialog(
-      child: new Column(
-        mainAxisSize: MainAxisSize.min,
-        crossAxisAlignment: CrossAxisAlignment.stretch,
-        children: children,
+    return new Padding(
+      padding: const EdgeInsets.symmetric(vertical: edgePadding),
```

Comment:
Is this for preventing the dialog box from touching the edge of the screen? Just wondering why vertical only.

Reply:
Yes, it's to keep it inset from the edge.  The dialog has a fixed width, so there's no reason to do it horizontally.